src/Features/LanguageServer/ProtocolUnitTests/References/FindAllReferencesHandlerTests.cs

```diff
@@ -241,14 +241,11 @@ void M()
 
             var results = await RunFindAllReferencesAsync<LSP.VSInternalReferenceItem>(testLspServer, testLspServer.GetLocations("caret").First());
 
-            // Namespace definitions should not have a location
-            Assert.True(results.Any(r => r.DefinitionText != null && r.Location == null));
-
-            // Namespace references should have a location
-            Assert.True(results.Any(r => r.DefinitionText == null && r.Location != null));
+            // Namespace source definitions and references should have locations:
+            Assert.True(results.All(r => r.Location != null));
 
             AssertValidDefinitionProperties(results, 0, Glyph.Namespace);
-            AssertHighlightCount(results, expectedDefinitionCount: 0, expectedWrittenReferenceCount: 0, expectedReferenceCount: 2);
+            AssertHighlightCount(results, expectedDefinitionCount: 1, expectedWrittenReferenceCount: 0, expectedReferenceCount: 2);
```

Comment:
```C#
namespace N
{
    class F
    {
        N.F f;
        global::N.F g;
    }
}

namespace N
{
}
```

Before:

FAR:
![image](https://github.com/dotnet/roslyn/assets/41759/b3e12dc0-cfb6-4ae7-ad22-adb45e09b9a0)

GTD:
![image](https://github.com/dotnet/roslyn/assets/41759/fdc22f06-d2be-4393-bbfa-a7b2123e469c)

After:

FAR:
![image](https://github.com/dotnet/roslyn/assets/41759/f82f8a6f-8e6f-4001-8a52-3be263f462c6)

GTD:
![image](https://github.com/dotnet/roslyn/assets/41759/0bbd3306-04ee-4fbf-bb81-56452db4ec02)



GTD:

Reply:
i do not love this.  but we can go with this for now and see if there is feedback.  

First: The primary difference here is that partial classes are normally very small in terms of partial count.  While namespaces are often *every file* in a particular project (or solution!).  So that means if someone is looking for the palce the namespace is referenced, this is now more painful.

Second: Can you test what happens if someone does goto-def on `global` in `global::System...`?  I want to maek sure that's ok :)

Reply:
> So that means if someone is looking for the palce the namespace is referenced, this is now more painful.

I'm not sure why would that be. Previously the references were mixed with definitions. Now FAR actually finds references, not definitions. The same happens for all symbols. If I invoke FAR on a method that's not called I get:

![image](https://github.com/dotnet/roslyn/assets/41759/c5b0c91b-9e0a-4758-b98c-ac642a1b0df5)

The definition is not included.

If I want all definitions of a namespace then I can now use GTD to find them. Previously it did not work.

Reply:
GTD on `global` shows 
![image](https://github.com/dotnet/roslyn/assets/41759/ac6df8ba-66bf-4f76-b2ad-b1a328349308)

Reply:
If you return a global namespace symbol from Semantic Search it displays `<global namespace>` just like you'd expect.